### PR TITLE
REP-358: new Replicator Schema Translation Demo

### PR DESCRIPTION
**Pull request by thomaskwscott**
https://confluentinc.atlassian.net/browse/REP-358

This is an example in preparation for a future DOCS ticket (DOCS-1333) that discusses Schema Translation outside Confluent Cloud.

### CODE REVIEW

**Review by ybyzek — CHANGES_REQUESTED** [review]
@thomaskwscott thanks for the PR.  As we discussed offline, let's move the README content to the docs flow, as done similarly for:

- https://github.com/confluentinc/examples/tree/5.3.1-post/ccloud/docs
- https://github.com/confluentinc/examples/tree/5.3.1-post/kubernetes/gke-base/docs
- https://github.com/confluentinc/cp-demo/tree/5.3.1-post/docs

Thanks in advance!

**Review by ybyzek — CHANGES_REQUESTED** [review]
* Missing `conf.py` (needs to be added into the `docs` folder)
* Update replicator-schema-translation/README: strip out the meat and just add a pointer to the docs landing page

File: replicator-schema-translation/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Replicator Schema Translation Demo
+
+## Overview
+
+Confluent Replicator features the ability to translate entries from a source Schema Registry to a destination Schema Registry.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
|crep-full| features the ability to translate entries from a source |sr| to a destination |sr|.
```

File: replicator-schema-translation/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+This demo provides a docker-compose environment with source and destination Schema Registries in which Schemas will be translated. In this demo we will create an entry in the source Schema Registry and translate it to the destination.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
This demo provides a docker-compose environment with source and destination Schema Registries in which schemas are translated. In this demo, you create an entry in the source |sr| and translate it to the destination.
```

File: replicator-schema-translation/README.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+The scripts directory provides examples of the operations that must be performed to prepare for the translation as well as JSON Replicator configurations required.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
The `scripts` directory provides examples of the operations that you must perform to prepare for the translation, as well as JSON Replicator configurations required.
```

File: replicator-schema-translation/README.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+## Prerequisites
+
+These demos require the following tools:
+
+* docker-compose
+
+These demos are memory intensive and Docker must be tuned accordingly. In Docker's advanced settings, increase the memory dedicated to Docker to at least 8GB (default is 2GB).
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
These demos are memory intensive and Docker must be tuned accordingly. In Docker's advanced settings, increase the memory dedicated to Docker to at least 8GB (the default is 2GB).
```

File: replicator-schema-translation/README.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+## Components
+
+In addition to Confluent Replicator, the environment contains:
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
In addition to |crep-full|, the environment contains:
```

File: replicator-schema-translation/README.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+* A single node source cluster containing srcKafka1, in production at least 3 nodes are recommended.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A single node source cluster containing `srcKafka1`. In production, at least 3 nodes are recommended.
```

File: replicator-schema-translation/README.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+* A single node destination cluster containing destKafka1, in production at least 3 nodes are recommended.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A single node destination cluster containing `destKafka1`. In production, at least 3 nodes are recommended.
```

File: replicator-schema-translation/README.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+* A single node Schema Registry cluster containing srcSchemaregistry, in production a master/standby configuration is recommended.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A single node Schema Registry cluster containing `srcSchemaregistry`. In production, a master/standby configuration is recommended.
```

File: replicator-schema-translation/README.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+* A single node Schema Registry cluster containing destSchemaregistry, in production a master/standby configuration is recommended.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A single node Schema Registry cluster containing `destSchemaregistry`. In production, a master/standby configuration is recommended.
```

File: replicator-schema-translation/README.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+* A single Connect worker that will run replicator
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A single Connect worker that runs Replicator.
```

File: replicator-schema-translation/README.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+* A source client container (kafkaClient) that creates a source topic and provides test data
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
* A source client container (`kafkaClient`) that creates a source topic and provides test data.
```

File: replicator-schema-translation/README.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+In all containers, minimal non-replicator configuration is supplied.
+
+## Setup and verifying the environments
+
+The demo performs the following steps:
+
+1. Creation of a subject in the source Schema Registry
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
1. Create a subject in the source Schema Registry.
```

File: replicator-schema-translation/README.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+2. Preparation for Schema translation
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
2. Prepare for schema translation.
```

File: replicator-schema-translation/README.md [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+3. Execution of Replicator to perform migration
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
3. Execute Replicator to perform migration.
```

File: replicator-schema-translation/README.md [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+4. Post translation Schema registry configuration.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
4. Post-translation Schema Registry configuration. 
```

File: replicator-schema-translation/README.md [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+2. Preparation for Schema translation
+3. Execution of Replicator to perform migration
+4. Post translation Schema registry configuration. 
+
+### Step 1: 
+
+This step is completed automatically when the environment is started. 
+
+Run:
+  ```
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]: Need to outdent these code blocks.

File: replicator-schema-translation/README.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+Verify the destination Schema Registry:
+  ```
+  curl http://localhost:8086/subjects
+  ```
+You should see the following:
+  ```
+  []
+  ```
+
+Note: the kafkaClient container must complete and exit 0 before verification
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Note: The `kafkaClient` container must complete and `exit 0` before verification.
```

File: replicator-schema-translation/README.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+You should see the following:
+  ```
+  []
+  ```
+
+Note: the kafkaClient container must complete and exit 0 before verification  
+
+### Step 2: 
+
+To prepare for Schema translation the source cluster must be put in "READONLY" mode and the destination in "IMPORT" mode.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
To prepare for schema translation, you must put the source cluster in "READONLY" mode and the destination cluster in "IMPORT" mode.
```

File: replicator-schema-translation/README.md [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+
+Run:
+  ```
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]: outdent

File: replicator-schema-translation/README.md [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+Verify: 
+
+This should produce:
+  ```
+  {"mode":"READONLY"}{"mode":"IMPORT"}
+  ```
+### Step 3: 
+
+Now we submit Replicator to perform the translation.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Submit Replicator to perform the translation.
```

File: replicator-schema-translation/README.md [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+
+Run:
+  ```
+  docker-compose exec connect /etc/kafka/scripts/submit_replicator.sh
+  ```
+
+Verify: 
+
+This should produce:
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Your output should resemble:
```

File: replicator-schema-translation/README.md [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+  ```
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
.. code:: json
```

File: replicator-schema-translation/README.md [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+  {"name":"testReplicator","config":{"connector.class":"io.confluent.connect.replicator.ReplicatorSourceConnector","topic.whitelist":"_schemas","topic.rename.format":"${topic}.replica","key.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","value.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","src.kafka.bootstrap.servers":"srcKafka1:10091","dest.kafka.bootstrap.servers":"destKafka1:11091","tasks.max":"1","confluent.topic.replication.factor":"1","schema.subject.translator.class":"io.confluent.connect.replicator.schemas.DefaultSubjectTranslator","schema.registry.topic":"_schemas","schema.registry.url":"http://destSchemaregistry:8086","name":"testReplicator"},"tasks":[],"type":"source"}
+  ```
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion

```

File: replicator-schema-translation/README.md [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+To prepare for Schema translation the source cluster must be put in "READONLY" mode and the destination in "IMPORT" mode.
+
+Run:
+  ```
+  docker-compose exec connect /etc/kafka/scripts/set_sr_modes_pre_translation.sh
+  ```
+
+Verify: 
+
+This should produce:
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Your output should resemble:
```

File: replicator-schema-translation/README.md [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+  ```
+  {"name":"testReplicator","config":{"connector.class":"io.confluent.connect.replicator.ReplicatorSourceConnector","topic.whitelist":"_schemas","topic.rename.format":"${topic}.replica","key.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","value.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","src.kafka.bootstrap.servers":"srcKafka1:10091","dest.kafka.bootstrap.servers":"destKafka1:11091","tasks.max":"1","confluent.topic.replication.factor":"1","schema.subject.translator.class":"io.confluent.connect.replicator.schemas.DefaultSubjectTranslator","schema.registry.topic":"_schemas","schema.registry.url":"http://destSchemaregistry:8086","name":"testReplicator"},"tasks":[],"type":"source"}
+  ```
+
+Note: If replicator is submitted before the Connect worker is ready then the message below will be printed. The script will automatically retry until successful
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Note: If replicator is submitted before the Connect worker is ready, the following message is printed. The script automatically retries until successful.
```

File: replicator-schema-translation/README.md [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+  Failed to submit replicator to Connect. This could be because the Connect worker is not yet started. Will retry in 10 seconds
+  ```
+
+### Step 4: 
+
+First verify the translation. 
+  ```
+  curl http://localhost:8086/subjects
+  ```
+You should see the following:
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Your output should resemble:
```

File: replicator-schema-translation/README.md [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+  ```
+  ["testTopic.replica-value"]
+  ```
+
+Note: Replicator may not immediately translate the Schema so this verification should be retired for up to 60 seconds.
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Note: Replicator may not translate the schema immediately, so this verification should be retried for up to 60 seconds.
```

File: replicator-schema-translation/README.md [hunk 28]
```diff
@@ -0,0 +1,132 @@
...
+
+Then run:
+  ```
+  docker-compose exec connect /etc/kafka/scripts/set_sr_modes_post_translation.sh
+  ```
+Verify: 
+
+This should produce:
```
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
Your output should resemble:
```

File: replicator-schema-translation/docs/index.rst [hunk 29]
```diff
@@ -0,0 +1,91 @@
+.. _quickstart-demos-replicator-schema-translation:
+
+Replicator Schema Translation Demo
+==================================
+
+This |crep| demo showcases the transfer of schemas stored in Confluent Schema Registry from one cluster to another.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
This |crep| demo showcases the transfer of schemas stored in |sr-long| from one cluster to another.
```

File: replicator-schema-translation/docs/index.rst [hunk 30]
```diff
@@ -0,0 +1,91 @@
...
+
+========
+Overview
+========
+
+Confluent Replicator features the ability to translate entries from a source Schema Registry to a destination Schema Registry.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
|crep| features the ability to translate entries from a source |sr| to a destination |sr|.
```

File: replicator-schema-translation/docs/index.rst [hunk 31]
```diff
@@ -0,0 +1,91 @@
...
+
+This demo provides a docker-compose environment with source and destination Schema Registries in which Schemas will be translated. In this demo we will create an entry in the source Schema Registry and translate it to the destination.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
This demo provides a docker-compose environment with source and destination registries in which schemas are translated. In this demo, you create an entry in the source |sr| and translate it to the destination.
```

File: replicator-schema-translation/docs/index.rst [hunk 32]
```diff
@@ -0,0 +1,91 @@
...
+
+The scripts directory provides examples of the operations that must be performed to prepare for the translation as well as JSON |crep| configurations required.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
The ``scripts`` directory provides examples of the operations that you must perform to prepare for the translation, as well as JSON |crep| configurations required.
```

File: replicator-schema-translation/docs/index.rst [hunk 33]
```diff
@@ -0,0 +1,91 @@
...
+
+=============
+Prerequisites
+=============
+
+**Demo validated with:**
+
+-  Confluent Platform |version|
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
-  |cp| |version|
```

File: replicator-schema-translation/docs/index.rst [hunk 34]
```diff
@@ -0,0 +1,91 @@
...
+-  Docker 19.03.2
+-  Docker-compose 1.24.1
+-  Java version 1.8.0_162
+-  MacOS 10.12
+
+1. Clone the `examples GitHub repository <https://github.com/confluentinc/examples>`__.
+
+   .. sourcecode:: bash
+
+     $ git clone https://github.com/confluentinc/examples
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
     git clone https://github.com/confluentinc/examples
```

File: replicator-schema-translation/docs/index.rst [hunk 35]
```diff
@@ -0,0 +1,91 @@
...
+
+2. Change directory to the Schema Translation demo.
+
+   .. sourcecode:: bash
+
+     $ cd examples/replicator-schema-translation
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
     cd examples/replicator-schema-translation
```

File: replicator-schema-translation/docs/index.rst [hunk 36]
```diff
@@ -0,0 +1,91 @@
...
+
+3. Start the entire demo by running a single command. This automatically creates source and destination clusters and adds a Schema to the source cluster Schema Registry. This will take less than 5 minutes to complete.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
3. Start the entire demo by running a single command that creates source and destination clusters automatically and adds a schema to the source cluster |sr|. This takes less than 5 minutes to complete.
```

File: replicator-schema-translation/docs/index.rst [hunk 37]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
+
+      $ docker-compose up -d
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
      docker-compose up -d
```

File: replicator-schema-translation/docs/index.rst [hunk 38]
```diff
@@ -0,0 +1,91 @@
...
+
+   Verify the demo has completely started by using Google Chrome to view the Schema Registries, the source cluster is at: http://localhost:8085/subjects (you should see: ``["testTopic-value"]``) and the destinartion cluster is at: http://localhost:8086/subjects (you should see: ``[]``).
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
   Verify the demo has completely started by using Google Chrome to view the Schema Registries. The source cluster is at: ``http://localhost:8085/subjects`` (you should see: ``["testTopic-value"]``) and the destination cluster is at: ``http://localhost:8086/subjects`` (you should see: ``[]``).
```

File: replicator-schema-translation/docs/index.rst [hunk 39]
```diff
@@ -0,0 +1,91 @@
...
+
+4. To prepare for Schema translation the source cluster Schema Registry must be put in "READONLY" mode and the destination in "IMPORT" mode.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
4. To prepare for schema translation, you must put the source cluster registry in "READONLY" mode and the destination registry in "IMPORT" mode.
```

File: replicator-schema-translation/docs/index.rst [hunk 40]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
+
+      $ docker-compose exec connect /etc/kafka/scripts/set_sr_modes_pre_translation.sh
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
      docker-compose exec connect /etc/kafka/scripts/set_sr_modes_pre_translation.sh
```

File: replicator-schema-translation/docs/index.rst [hunk 41]
```diff
@@ -0,0 +1,91 @@
...
+
+   This should produce:
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
   Your output should resemble:
```

File: replicator-schema-translation/docs/index.rst [hunk 42]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
   .. sourcecode:: json
```

File: replicator-schema-translation/docs/index.rst [hunk 43]
```diff
@@ -0,0 +1,91 @@
...
+
+      {"mode":"READONLY"}{"mode":"IMPORT"}
+
+5. Now we submit Replicator to perform the translation.
+
+   .. sourcecode:: bash
+
+      $ docker-compose exec connect /etc/kafka/scripts/submit_replicator.sh
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
      docker-compose exec connect /etc/kafka/scripts/submit_replicator.sh
```

File: replicator-schema-translation/docs/index.rst [hunk 44]
```diff
@@ -0,0 +1,91 @@
...
+
+      $ docker-compose exec connect /etc/kafka/scripts/set_sr_modes_pre_translation.sh
+
+   This should produce:
+
+   .. sourcecode:: bash
+
+      {"mode":"READONLY"}{"mode":"IMPORT"}
+
+5. Now we submit Replicator to perform the translation.
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
5. Submit |crep| to perform the translation.
```

File: replicator-schema-translation/docs/index.rst [hunk 45]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
+
+      $ docker-compose exec connect /etc/kafka/scripts/submit_replicator.sh
+
+   This should produce:
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
   Your output should resemble:
```

File: replicator-schema-translation/docs/index.rst [hunk 46]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
   .. sourcecode:: json
```

File: replicator-schema-translation/docs/index.rst [hunk 47]
```diff
@@ -0,0 +1,91 @@
...
+
+      {"name":"testReplicator","config":{"connector.class":"io.confluent.connect.replicator.ReplicatorSourceConnector","topic.whitelist":"_schemas","topic.rename.format":"${topic}.replica","key.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","value.converter":"io.confluent.connect.replicator.util.ByteArrayConverter","src.kafka.bootstrap.servers":"srcKafka1:10091","dest.kafka.bootstrap.servers":"destKafka1:11091","tasks.max":"1","confluent.topic.replication.factor":"1","schema.subject.translator.class":"io.confluent.connect.replicator.schemas.DefaultSubjectTranslator","schema.registry.topic":"_schemas","schema.registry.url":"http://destSchemaregistry:8086","name":"testReplicator"},"tasks":[],"type":"source"}
+
+6. Now verify the translation by revisiting the Schema Registries at: http://localhost:8085/subjects and: http://localhost:8086/subjects. They should both now list schemas with the destination cluster showing ``["testTopic.replica-value"]``:
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
6. Now verify the translation by revisiting the registries at: ``http://localhost:8085/subjects`` and ``http://localhost:8086/subjects``. They should both list schemas with the destination cluster showing ``["testTopic.replica-value"]``.
```

File: replicator-schema-translation/docs/index.rst [hunk 48]
```diff
@@ -0,0 +1,91 @@
...
+
+7. To complete the demo return both Schema Registries to ``READWRITE`` mode:
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
7. To complete the demo return both clusters to ``READWRITE`` mode:
```

File: replicator-schema-translation/docs/index.rst [hunk 49]
```diff
@@ -0,0 +1,91 @@
...
+
+   .. sourcecode:: bash
+
+      $ docker-compose exec connect /etc/kafka/scripts/set_sr_modes_post_translation.sh
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
      docker-compose exec connect /etc/kafka/scripts/set_sr_modes_post_translation.sh
```

File: replicator-schema-translation/docs/index.rst [hunk 50]
```diff
@@ -0,0 +1,91 @@
...
+
+========
+Teardown
+========
+
+1. Stop the demo, destroy all local components.
+
+   .. sourcecode:: bash
+
+      $ docker-compose down
```
Comment (JimGalasyn) [inline replicator-schema-translation/docs/index.rst]:
```suggestion
      docker-compose down
```

File: replicator-schema-translation/README.md (hunk 2 above)
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
This demo provides a docker-compose environment with source and destination Schema Registries in which schemas are translated. In this demo, you create an entry in the source Schema Registry and translate it to the destination.
```

File: replicator-schema-translation/README.md (hunk 3 above)
Comment (JimGalasyn) [inline replicator-schema-translation/README.md]:
```suggestion
The `scripts` directory provides examples of the operations that you must perform to prepare for the translation, as well as JSON Replicator configurations required.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with some copy edits!

**Comment by thomaskwscott** [comment]
> * Missing `conf.py` (needs to be added into the `docs` folder)
> * Update replicator-schema-translation/README: strip out the meat and just add a pointer to the docs landing page

conf.py added. Can we leave the Readme until we've sorted where this will go in the docs?

**Comment by ybyzek** [comment]
> Can we leave the Readme until we've sorted where this will go in the docs?

1. What's required to sort out where this will go in docs?  Can that be done in parallel?
2. Once removing content from the README and merging, there will be just a 20 minute window where it's not available.  IMO that's an acceptable downtime when there are no public docs

**Comment by thomaskwscott** [comment]
> > Can we leave the Readme until we've sorted where this will go in the docs?
> 
> 1. What's required to sort out where this will go in docs?  Can that be done in parallel?
> 2. Once removing content from the README and merging, there will be just a 20 minute window where it's not available.  IMO that's an acceptable downtime when there are no public docs

Good point, I just had a look and the associated docs ticket is small so we should be able to do these simultaneously. I've changed the readme as requested.

File: replicator-schema-translation/README.md [hunk 51]
```diff
@@ -8,125 +8,24 @@ This demo provides a docker-compose environment with source and destination Sche
...
 
-Your output should resemble:
-  ```
-  {"mode":"READWRITE"}{"mode":"READWRITE"}
-  ```
+These demos are memory intensive and Docker must be tuned accordingly. In Docker's advanced settings, increase the memory dedicated to Docker to at least 8GB (the default is 2GB).
 
+# Running the demo
 
+You can find the documentation for running this demo at [https://docs.confluent.io/current/tutorials/examples/ccloud/docs/index.html](https://docs.confluent.io/current/tutorials/examples/replicator-schema-translation/docs/index.html)
```
Comment (ybyzek) [inline replicator-schema-translation/README.md]: Is there an error in the link `https://docs.confluent.io/current/tutorials/examples/ccloud/docs/index.html`?

File: replicator-schema-translation/README.md [hunk 52]
```diff
@@ -8,125 +8,24 @@ This demo provides a docker-compose environment with source and destination Sche
...
-
-First verify the translation. 
-  ```
-  curl http://localhost:8086/subjects
-  ```
-Your output should resemble:
-  ```
-  ["testTopic.replica-value"]
-  ```
+## Prerequisites
```
Comment (ybyzek) [inline replicator-schema-translation/README.md]: For consideration: move Prereqs to the docs (not in the README) to simplify the README

**Review by ybyzek — APPROVED** [review]
Looking so good.  Two more nits but then ship it ;)

**Comment by thomaskwscott** [comment]
@ybyzek I fixed the remaining things, but I have no merge perms. Can you merge for me?

**Pull request merged**